src/lib.rs

```diff
@@ -1,70 +1,80 @@
-use image::{imageops::overlay, load_from_memory, ImageError, RgbaImage};
-use rand::random;
+use image::{imageops::overlay, ImageError, DynamicImage};
 use std::path::Path;
+use rand::prelude::*;
 
 mod emoji;
 
-fn sums_chunked(samples_a: &[u8], samples_b: &[u8]) -> (i32, i32, i32) {
-    samples_a
-        .chunks_exact(3)
-        .zip(samples_b.chunks_exact(3))
-        .fold((0, 0, 0), |(r, g, b), (p_a, p_b)| {
-            (
-                r + (p_a[0] as i32 - p_b[0] as i32).abs(),
-                g + (p_a[1] as i32 - p_b[1] as i32).abs(),
-                b + (p_a[2] as i32 - p_b[2] as i32).abs(),
-            )
+
+fn measure_dist_chunks(samples_a: &[u8], samples_b: &[u8]) -> i64 {
+    samples_a.iter()
+        .zip(samples_b.iter())
+        .fold(0, |rgba, (p_a, p_b)| {
+                rgba + (*p_a as i64 - *p_b as i64).abs()
         })
 }
 
+fn subimage_compare(image_a: &DynamicImage, image_b: &DynamicImage, x: u32, y: u32 ) -> i64 {
+    let sub_image_a = image_a.crop_imm(x,y,16,16).to_rgb();
+    let sub_image_b = image_b.crop_imm(x,y,16,16).to_rgb();
+    measure_dist_chunks(&sub_image_a, &sub_image_b)
+}
+
 pub fn generate_image(
-    image_buffer: &[u8],
+    image_buffer: &DynamicImage,
```

Comment:
this should get reverted, unless theres a reason to do it this way instead of accepting a byte array

Reply:
mainly because im not confident using a `DynamicImage` will work in wasm. i'm gonna try and build this in wasm tonight and i'll report back

Reply:
An image buffer is used for the crop method to get a sub image. Do you think that type in general is a problem or just as an input?

Reply:
If it's just an input problem then we could convert after

Reply:
If it's just an input problem then we could convert after

Reply:
If it's just an input problem then we could convert after

Reply:
i was thinking just as an input, but i looked into it and i don'
t think it will end up being an issue